vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -0,0 +1,261 @@
+#!/bin/bash
+
+testFilesDownloaded() {
+  test="testFilesDownloaded"
+  echo "$test:Start"
+  filesToDownload=$1
+
+  filesToDownload=$(echo $filesToDownload | jq -r ".[]" | jq . --monochrome-output --compact-output)
+  emptyFiles=()
+  missingPaths=()
+  for fileToDownload in ${filesToDownload[*]}; do
+    fileName=$(echo "${fileToDownload}" | jq .fileName -r)
+    downloadLocation=$(echo "${fileToDownload}" | jq .downloadLocation -r)
+    versions=$(echo "${fileToDownload}" | jq .versions -r | jq -r ".[]")
+
+    if [ ! -d $downloadLocation ]; then
+      err $test "Directory ${downloadLocation} does not exist"
+      missingPaths+=("$downloadLocation")
+      continue
+    fi
+
+    for version in ${versions}; do
+      file_Name=$(string_replace $fileName $version)
+      dest="$downloadLocation/${file_Name}"
+
+      if [ ! -s $dest ]; then
```

Comment:
Downloading the file and compare MD5 is time-consuming, but we can compare the size of remote file with the local file. 
`curl -sI $URL | grep -i Content-Length | awk '{print $2}'` will get the file size from the response header without downloading the file

Reply:
sure working on it

Reply:
done